haoshiyou/app/pages/listings-tab/listings-tab.page.ts

```diff
@@ -69,17 +68,50 @@ export class ListingsTabPage implements OnInit, OnDestroy {
     });
   }
 
+
   private updateMarkers() {
     if(this.mapReady && this.listings) {
       this.markers = [];
       for (let listing of this.listings) {
-        let opt:google.maps.MarkerOptions = <google.maps.MarkerOptions>{};
-        opt.position = new google.maps.LatLng(listing.lat, listing.lng);
-        opt.animation = google.maps.Animation.DROP;
-        opt.map = this.map;
-        let marker = new google.maps.Marker(opt);
+        //let opt:google.maps.MarkerOptions = <google.maps.MarkerOptions>{};
+        //opt.position = new google.maps.LatLng(listing.lat, listing.lng);
+        //opt.animation = google.maps.Animation.DROP;
+        //opt.map = this.map;
+        //let marker = new google.maps.Marker(opt);
+
+        //MarkerWithLabel.prototype = new google.maps.OverlayView();
+        //MarkerWithLabel.prototype.onRemove= function(){}
+
+
+        var priceConent= listing.price === undefined ? "<div class='arrow'></div><div class='inner'></div>" : "<div class='arrow'></div><div class='inner'>" + listing.price + "</div>";
```

Comment:
1. please split line for readability, better at 100 chars per line.
2. this line is composing an HTML component, you will need to sanitize "price" to avoid XSS attack